core/tests/webdriverio_utils/ContributorDashboardAdminPage.js

```diff
@@ -232,7 +234,94 @@ var ContributorDashboardAdminPage = function () {
       `.e2e-test-language-selector-option=${language}`
     );
     await action.click(`${language} option selector`, selectorOption);
-    await action.click('Language Selector', languageSelector);
+  };
+
+  this.selectDate = async function (datePickerToggle, day, month, year) {
+    await action.click('Date Picker Toggle', $(datePickerToggle));
+
+    if (year) {
+      var nextMonthButton = $('.mat-calendar-next-button');
+      var prevMonthButton = $('.mat-calendar-previous-button');
```

Comment:
@U8NWXD, I think adding `.e2e-*` here would not be a good idea as this datepicker component comes inbuilt with angular material package, Alternatively, we can use full XPath of the element, for eg. `//body/div[1]/div[2]/div/mat-datepicker-content/div[2]/mat-calendar/mat-calendar-header/div/div/button[2]` instead of `.mat-calendar-previous-button`.

Reply:
I don't think we should use the full XPath component, but can you find the first node in the DOM that's an ancestor to the desired node and under our control, add an `e2e-*` class to it, and use it as the target of the first selector in a chain? Then you can use your `.mat-calendar-*` classes in the second selector of each chain.

The reason we care about `e2e-*` selectors is that they help communicate to FE developers what elements in the DOM the E2E tests rely on, and therefore what changes will require fixes to the E2E tests. This also makes it easier for the dev workflow team to identify the relevant frontend code when debugging E2E tests.

Reply:
@U8NWXD, I tried adding `e2e-*` to a class of an ancestor to the desired node. However, whenever date picker is toggled an invisible overlay covers entire screen on with date picker in displayed, and the ancestor to this is the body tag itself in the DOM.
![Screenshot (151)](https://github.com/oppia/oppia/assets/92575005/3693f0b9-4387-4d5c-b678-0ce1bf5e6de4)

Reply:
Okay, in this case then I think we can make an exception to our usual `e2e-*` class rule. Can you add a comment explaining why can't use `e2e-*` class selectors here? That way future developers don't copy this pattern

Reply:
done